### PR TITLE
Switch travis to docker; use slim image

### PR DESCRIPTION
**Pull request by beaugunderson**
Some tests succeed but most currently fail with:

```
E               {"contracts":{},"errors":[{"component":"general","formattedMessage":"Cannot import url (\"/app/src/Controller.sol\"): File outside of allowed directories.","message":"Cannot import url (\"/app/src/Controller.sol\"): File outside of allowed directories.","severity":"error","type":"IOError"}],"sources":{}}
```

Does the current image have the same issue?

### CODE REVIEW

**Comment by MicahZoltu** [comment]
Rebase on the recent PR merge, should resolve that.

**Comment by beaugunderson** [comment]
@MicahZoltu only one failure now; I think this is ready for a merge:

https://travis-ci.org/AugurProject/augur-core/builds/256187053

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,23 @@
 dist: trusty
-sudo: false
+sudo: required
 
-cache:
-  - apt
-  - pip
+# cache:
+#   directories:
+#     - $HOME/docker
```
Comment (MicahZoltu) [inline .travis.yml]: Recommend deleting rather than commenting out.

File: Dockerfile-test [hunk 2]
```diff
@@ -1,42 +1,24 @@
...
-	&& pip install -r requirements.txt \
-	&& apk del gcc g++ openssl-dev \
-	&& rm -rf /var/cache/apk/*
 
-# install runtime dependencies
-RUN apk --no-cache --update add libstdc++ libssl1.0 \
-	&& rm -rf /var/cache/apk/*
+RUN cd /app && \
+  pip install --upgrade pip setuptools && \
+  pip install -r requirements.txt
```
Comment (MicahZoltu) [inline Dockerfile-test]: I prefer tabs over spaces (I thought I had an .editorconfig to enforce this, but apparently not).

File: Dockerfile-test [hunk 3]
```diff
@@ -1,42 +1,24 @@
...
-	&& wget -qO- https://github.com/ethereum/solidity/releases/download/v0.4.13/solidity_0.4.13.tar.gz | tar -xvz -C /solidity --strip 1 \
-	&& cd /solidity && cmake -DCMAKE_BUILD_TYPE=Release -DTESTS=0 -DSTATIC_LINKING=1 \
-	&& cd /solidity && make solc && install -s  solc/solc /usr/bin \
-	&& cd / && rm -rf solidity \
-	&& apk del sed build-base git make cmake gcc g++ musl-dev curl-dev boost-dev \
-	&& rm -rf /var/cache/apk/*
+FROM python:2.7.13
+
+RUN wget --quiet --output-document /usr/local/bin/solc \
+  https://github.com/ethereum/solidity/releases/download/v0.4.13/solc-static-linux && \
```
Comment (MicahZoltu) [inline Dockerfile-test]: I prefer `&&` at the start of the line so it is clear to the read that this is a continuation, without them having to read to the end of the line to find out.  Helps quickly identify bugs where an `&&` is missing which can change behavior (but not fail) in some cases.

Comment (MicahZoltu) [inline Dockerfile-test]: Also recommend keeping each command on one line, not split across lines.  In this case, this line is part of the previous line so there should not be a line break here.

File: .dockerignore [hunk 4]
```diff
@@ -2,4 +2,4 @@
 **/*.pyc
 **/.cache/
 **/compilation_cache/
-src/macros/externs.sem
```
Comment (MicahZoltu) [inline .dockerignore]: Ah, totally forgot to remove this when I removed the `generateExterns` call.  My bad.

**Comment by beaugunderson** [comment]
@MicahZoltu 👍 made your suggested changes

**Pull request merged**